### PR TITLE
bouncing bettys the unbreaking

### PR DESCRIPTION
**Pull request by johnravioliUSA**
## About The Pull Request
(This PR is a carbon copy of #773 but instead of being batshit broken on my end, this is more up to date. my bad for making a commit 3 months ago and never using it LOL ok heres the updated one)

This PR adds in a new landmine, exclusively able to be bought from squad engineer vendors, or combat technician vendors for 30 points. It can be obtained through requisitions, although a limited amount exist there.
Requisitions has 2 at the start of a round, but it scales with the pop, so 2 on low, 4 on high. This is because it is effectively a better claymore, so that's why it's halved.

The mine itself is based off the bouncing betty, a mostly shrapnel based mine, which is what I replicated. Once detonated, the bounding mine has 48 pieces of shrapnel that shoot out compared to the claymore's 12 in one direction. This is prone to changes as it might be too powerful, who knows. The mine itself is capable of dealing massive amounts of damage to back liners, ideally runners or lurkers. It can crit them if they get RNG'd, but there is basically no way of it insta killing unless you're wounded already.

You can only plant these on actual ground, not the floor of colony buildings or the Almayer itself. There is a 50/50 chance of when you disarm it that you will blow up, this is only when you don't have the faction's IFF (I.e. you are CLF disarming a USCM mine).

Demonstration: the pause in the video is a runtime error, which probably effected the explosion. Besides that you'd have to see that for yourself.

https://streamable.com/e2m85b

https://streamable.com/e9mcuj (Older video with older stats, that were changed. imagine this but smaller, like 50% the size of this shit)

## Why It's Good For The Game

The bouncing betty is canon in Aliens lore, as it is a marine landmine. Besides that, I believe that more variety in some of the items is better, because the claymore isn't the only landmine that a marine force would use. I think this would go nicely for something like backline defense, defending the FOB from being harassed by lurkers and runners by strategically putting landmines around in the jungle, or desert, whatever the case may be. Would probably make for some interesting gameplay, that's all I got to say.
Would also make HvH more hectic with more things you can give marines AND UPP / CLF

## Changelog

:cl:
add: The M5 Bounding Mine aka Bouncing Betties. These mines can be obtained from req or engineer prep vendors. The mine itself is a low power shrapnel landmine that shoots out a lot of shrapnel in a 360 degree angle when you step on it. Sprites by Frans_Feiffer
/:cl:

### CODE REVIEW

**Comment by sose96** [comment]
I hope this makes it in, I'd love to make some minefield nightmare inserts.

**Review by Agameofscones — APPROVED** [review]
sprites lgtm

File: ColonialMarinesALPHA.dme [hunk 1]
```diff
@@ -81,10 +81,10 @@
 #include "code\__DEFINES\stamina.dm"
 #include "code\__DEFINES\STUI.dm"
 #include "code\__DEFINES\subsystem.dm"
+#include "code\__DEFINES\text.dm"
 #include "code\__DEFINES\surgery.dm"
 #include "code\__DEFINES\techtree.dm"
 #include "code\__DEFINES\tests.dm"
-#include "code\__DEFINES\text.dm"
```
Comment (stanalbatross) [inline ColonialMarinesALPHA.dme:87]: keep in original place

File: ColonialMarinesALPHA.dme [hunk 2]
```diff
@@ -251,8 +251,8 @@
 #include "code\datums\_atmos_setup.dm"
 #include "code\datums\action.dm"
 #include "code\datums\ASRS.dm"
-#include "code\datums\beam.dm"
 #include "code\datums\browser.dm"
+#include "code\datums\beam.dm"
```
Comment (stanalbatross) [inline ColonialMarinesALPHA.dme:255]: keep in original place

File: code/game/machinery/vending/vendor_types/squad_prep/squad_engineer.dm [hunk 3]
```diff
@@ -38,6 +38,7 @@ GLOBAL_LIST_INIT(cm_vending_gear_engi, list(
 		list("M74 AGM-Smoke Airburst Packet (x3 airburst grenades)", 10, /obj/item/storage/box/packet/airburst_smoke, null, VENDOR_ITEM_REGULAR),
 		list("M74 AGM-Hornet Airburst Packet (x3 airburst grenades", 20, /obj/item/storage/box/packet/hornet, null, VENDOR_ITEM_REGULAR),
 		list("M20 Mine Box (x4 mines)", 18, /obj/item/storage/box/explosive_mines, null, VENDOR_ITEM_REGULAR),
+		list("M5 Mine Box (x4 mines)", 30, /obj/item/storage/box/explosive_mines/bounding, null, VENDOR_ITEM_REGULAR),
 		list("M40 MFHS Metal Foam Grenade", 5, /obj/item/explosive/grenade/metal_foam, null, VENDOR_ITEM_REGULAR),
```
Comment (stanalbatross) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_engineer.dm:42]: 5 comes after 4

File: code/game/objects/items/explosives/mine.dm [hunk 4]
```diff
@@ -312,3 +313,113 @@
...
+	if(check_for_obstacles(user))
+		return
+
+	if(active || user.action_busy)
+		return
+
+	user.visible_message(SPAN_NOTICE("[user] starts deploying [src]."), \
+		SPAN_NOTICE("You start deploying [src]."))
+	if(!do_after(user, 4 SECONDS, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE))
+		user.visible_message(SPAN_NOTICE("[user] stops deploying [src]."), \
```
Comment (stanalbatross) [inline code/game/objects/items/explosives/mine.dm:346]: \the [src]

File: code/game/objects/items/explosives/mine.dm [hunk 5]
```diff
@@ -312,3 +313,113 @@
...
+		return
+
+	if(active)
+		return
+
+	if(check_for_obstacles(user))
+		return
+
+	user.visible_message(SPAN_NOTICE("[user] finishes deploying [src]."), \
+		SPAN_NOTICE("You finish deploying [src]."))
```
Comment (stanalbatross) [inline code/game/objects/items/explosives/mine.dm:357]: \the

File: code/game/objects/items/explosives/mine.dm [hunk 6]
```diff
@@ -312,3 +313,113 @@
...
+	cause_data = create_cause_data(initial(name), user)
+	anchored = TRUE
+	playsound(loc, 'sound/weapons/mine_armed.ogg', vol = 25, vary = TRUE)
+	user.drop_inv_item_on_ground(src)
+	activate_sensors()
+	update_icon()
+
+/obj/item/explosive/mine/bounding/check_for_obstacles(mob/living/user)
+	if(user.loc && (user.loc.density || is_mainship_level(user.z)))
+		to_chat(user, SPAN_WARNING("You can't plant a mine here."))
```
Comment (stanalbatross) [inline code/game/objects/items/explosives/mine.dm:371]: \a [src]

File: code/game/objects/items/explosives/mine.dm [hunk 7]
```diff
@@ -312,3 +313,113 @@
...
+		return TRUE
+	if(user.z == GLOB.interior_manager.interior_z)
+		to_chat(user, SPAN_WARNING("It's too cramped in here to deploy \a [src]."))
+		return TRUE
+	if(locate(/obj/item/explosive/mine) in get_turf(src))
+		to_chat(user, SPAN_WARNING("There already is a mine at this position!"))
```
Comment (stanalbatross) [inline code/game/objects/items/explosives/mine.dm:377]: is already a mine

File: code/game/objects/items/explosives/mine.dm [hunk 8]
```diff
@@ -312,3 +313,113 @@
...
+			if(user.action_busy)
+				return
+			if(user.faction == iff_signal)
+				user.visible_message(SPAN_NOTICE("[user] starts disarming [src]."), \
+				SPAN_NOTICE("You start disarming [src]."))
+			else
+				user.visible_message(SPAN_NOTICE("[user] starts fiddling with \the [src], trying to disarm it."), \
+				SPAN_NOTICE("You start disarming [src], but you don't know its IFF data. This might end badly..."))
+			if(!do_after(user, 3 SECONDS, INTERRUPT_NO_NEEDHAND, BUSY_ICON_FRIENDLY))
+				user.visible_message(SPAN_WARNING("[user] stops disarming [src]."), \
```
Comment (stanalbatross) [inline code/game/objects/items/explosives/mine.dm:393]: \the [src]

File: code/game/objects/items/explosives/mine.dm [hunk 9]
```diff
@@ -312,3 +313,113 @@
...
+		if(active)
+			if(user.action_busy)
+				return
+			if(user.faction == iff_signal)
+				user.visible_message(SPAN_NOTICE("[user] starts disarming [src]."), \
+				SPAN_NOTICE("You start disarming [src]."))
+			else
+				user.visible_message(SPAN_NOTICE("[user] starts fiddling with \the [src], trying to disarm it."), \
+				SPAN_NOTICE("You start disarming [src], but you don't know its IFF data. This might end badly..."))
+			if(!do_after(user, 3 SECONDS, INTERRUPT_NO_NEEDHAND, BUSY_ICON_FRIENDLY))
```
Comment (stanalbatross) [inline code/game/objects/items/explosives/mine.dm:392]: should probably be interrupt all

File: code/game/objects/items/explosives/mine.dm [hunk 10]
```diff
@@ -312,3 +313,113 @@
...
+					triggered = TRUE
+					if(tripwire)
+						var/direction = reverse_dir[src.dir]
+						var/step_direction = get_step(src, direction)
+						tripwire.forceMove(step_direction)
+					prime()
+			if(!active)
+				return
+			user.visible_message(SPAN_NOTICE("[user] finishes disarming [src]."), \
+			SPAN_NOTICE("You finish disarming [src]."))
```
Comment (stanalbatross) [inline code/game/objects/items/explosives/mine.dm:407]: \the [src]

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Comment by PhantomEpicness** [comment]
seems like there is a lot of unnecessary copy and pasting of code where existing code could have simply been reworked to be more modular and therefore easier to make new mines in the future.

 I also happened to be working on a project like this in the past, check out PhantomEpicness/AntiTankMine for an example of a rework, feel free to copy from there if you want.

**Comment by johnravioliUSA** [comment]
I wasn't intending to rework something, just adding something that seemed cool. But I will take a look at it nethertheless

File: code/game/objects/items/storage/boxes.dm [hunk 11]
```diff
@@ -514,6 +514,18 @@
...
 
+/obj/item/storage/box/explosive_mines/bounding
+	name = "\improper M5 mine box"
+	desc = "A secure box holding four M5 anti-personel proximity mines."
+	icon_state = "bettybox"
+	w_class = SIZE_MEDIUM
+	max_storage_space = 8
+	can_hold = list(/obj/item/explosive/mine/bounding)
+
+/obj/item/storage/box/explosive_mines/fill_preset_inventory()
```
Comment (stanalbatross) [inline code/game/objects/items/storage/boxes.dm:525]:
```suggestion
/obj/item/storage/box/explosive_mines/bounding/fill_preset_inventory()
```

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Review by stanalbatross — CHANGES_REQUESTED** [review]
make it so that you can order them from ASRS for twice the price of regular claymores

**Comment by johnravioliUSA** [comment]
> make it so that you can order them from ASRS for twice the price of regular claymores

done

File: code/game/objects/items/explosives/mine.dm [hunk 12]
```diff
@@ -312,3 +313,113 @@
...
+
+//disarming, it's a little different than a normal claymore
+/obj/item/explosive/mine/bounding/attackby(obj/item/item, mob/user)
+	if(HAS_TRAIT(item, TRAIT_TOOL_WIRECUTTERS))
+		if(active)
+			if(user.action_busy)
+				return
+			if(user.faction == iff_signal)
+				user.visible_message(SPAN_NOTICE("[user] starts disarming \the [src]."), \
+				SPAN_NOTICE("You start disarming [src]."))
```
Comment (stanalbatross) [inline code/game/objects/items/explosives/mine.dm:388]:
```suggestion
				SPAN_NOTICE("You start disarming \the [src]."))
```

**Review by monkeysfist101 — CHANGES_REQUESTED** [review]
While the icons are fine on their own, they don't fit the setting. Please alter the appearance, name, and description, to reflect the G2 Electroshock Grenade (Sonic Electronic Ball Breakers)

The mechanics are fine, but please modify the explosion effects as well. I'll look into getting you the sound effects.

Thanks!

https://avp.fandom.com/wiki/G2_Electroshock_Grenade

**Comment by johnravioliUSA** [comment]
> While the icons are fine on their own, they don't fit the setting. Please alter the appearance, name, and description, to reflect the G2 Electroshock Grenade (Sonic Electronic Ball Breakers)
> 
> The mechanics are fine, but please modify the explosion effects as well. I'll look into getting you the sound effects.
> 
> Thanks!
> 
> https://avp.fandom.com/wiki/G2_Electroshock_Grenade

I will do this sooner or later, because I'm busy and not able to do them like I did before (i was on summer break). Expect it to be done within a week or so.

**Comment by monkeysfist101** [comment]
> > While the icons are fine on their own, they don't fit the setting. Please alter the appearance, name, and description, to reflect the G2 Electroshock Grenade (Sonic Electronic Ball Breakers)
> > The mechanics are fine, but please modify the explosion effects as well. I'll look into getting you the sound effects.
> > Thanks!
> > https://avp.fandom.com/wiki/G2_Electroshock_Grenade
> 
> I will do this sooner or later, because I'm busy and not able to do them like I did before (i was on summer break). Expect it to be done within a week or so.

Awesome. Just let me know how I can help.

**Comment by PhantomEpicness** [comment]
Added to list of PRs I will revive at a later date.

Will include this into my AT mine PR whenever I feel like working on it again.

**Pull request closed without merging**